lib/src/front_end/piece_writer.dart

```diff
@@ -302,8 +302,6 @@ class PieceWriter {
   void _flushSpace() {
     if (!_pendingSpace) return;
 
-    // TODO(perf): See if we can make SpacePiece a constant to avoid creating
-    // multiple.
```

Comment:
I tried this optimization a while back and it actually made things slower. (I'm guessing because it required `SpacePiece` to implement `Piece` instead of extending it and that in turn made a bunch of calls on `Piece` methods polymorphic?)